speechbrain/nnet/attention.py

```diff
@@ -938,3 +939,346 @@ def forward(self, x):
         x = x.permute(1, 0, 2)
 
         return x
+
+
+class RotationMatrix(nn.Module):
+    """
+    The rotation matrice for the :class:`~RoPEMHA`.
+    The shape of these matrices is supposedly to be max_lengh * d_attnhead * d_attnhead.
+    However, due to the sparsity of the rotation matrices, we only need to store max_lengh * d_attnhead elements
+    of the matrix.
+    see eq(15) of https://arxiv.org/pdf/2104.09864
+
```

Comment:
Add a docstring example please.  We may want to unittest that as well, see the unitary tests that we already have!